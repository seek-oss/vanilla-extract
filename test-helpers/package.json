{
  "name": "test-helpers",
  "private": true,
  "version": "0.0.0",
  "main": "src/index.ts",
  "bin": "src/bin.ts",
  "author": "SEEK",
  "license": "MIT",
  "dependencies": {
    "@babel/core": "^7.13.10",
    "@types/mini-css-extract-plugin": "^1.2.2",
    "@types/webpack-dev-server": "^3.11.1",
    "@vanilla-extract/esbuild-plugin": "*",
    "@vanilla-extract/webpack-plugin": "*",
    "babel-loader": "^8.2.2",
    "css-loader": "^5.1.3",
    "got": "^11.8.2",
    "html-webpack-plugin": "^5.2.0",
    "mini-css-extract-plugin": "^1.3.9",
    "minimist": "^1.2.5",
    "path-browserify": "^1.0.1",
    "portfinder": "^1.0.28",
    "style-loader": "^2.0.0",
    "webpack": "^5.27.1",
    "webpack-dev-server": "^3.11.2",
    "webpack-merge": "^5.7.3"
  },
  "devDependencies": {
    "@types/minimist": "^1"
  }
}
