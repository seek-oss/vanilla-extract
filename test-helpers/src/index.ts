export * from './startFixture';
export * from './getStylesheet';

export const getTestNodes = (fixture: string) =>
  require(`@fixtures/${fixture}/test-nodes.json`);
