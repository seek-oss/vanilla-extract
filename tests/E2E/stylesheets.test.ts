import { getStylesheet, startFixture, TestServer } from 'test-helpers';

const fixtures = ['unused-modules', 'themed'];

describe('Stylesheet', () => {
  describe.each(fixtures)('%s', (fixture) => {
    let server: TestServer;

    beforeAll(async () => {
      server = await startFixture(fixture, {
        type: 'mini-css-extract',
        mode: 'production',
        basePort: 9000,
      });
    });

    it('should create valid stylesheet', async () => {
      expect(await getStylesheet(server.url)).toMatchSnapshot();
    });

    afterAll(async () => {
      await server.close();
    });
  });
});
