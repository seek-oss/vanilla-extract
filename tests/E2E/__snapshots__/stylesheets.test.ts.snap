// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stylesheet sprinkles - esbuild should create valid stylesheet 1`] = `
".styles_display_flex_mobile__1j5zl920{display:flex}.styles_display_none_mobile__1j5zl924{display:none}.styles_display_block_mobile__1j5zl928{display:block}.styles_paddingTop_small_mobile__1j5zl92c{padding-top:10px}.styles_paddingTop_medium_mobile__1j5zl92g{padding-top:20px}@media screen and (min-width: 768px){.styles_display_flex_tablet__1j5zl921{display:flex}.styles_display_none_tablet__1j5zl925{display:none}.styles_display_block_tablet__1j5zl929{display:block}.styles_paddingTop_small_tablet__1j5zl92d{padding-top:10px}.styles_paddingTop_medium_tablet__1j5zl92h{padding-top:20px}}@media screen and (min-width: 1024px){.styles_display_flex_desktop__1j5zl922{display:flex}@supports not (display: grid){[data-dark-mode] .styles_display_flex_darkDesktop__1j5zl923{display:flex}[data-dark-mode] .styles_display_none_darkDesktop__1j5zl927{display:none}[data-dark-mode] .styles_display_block_darkDesktop__1j5zl92b{display:block}[data-dark-mode] .styles_paddingTop_small_darkDesktop__1j5zl92f{padding-top:10px}[data-dark-mode] .styles_paddingTop_medium_darkDesktop__1j5zl92j{padding-top:20px}}.styles_display_none_desktop__1j5zl926{display:none}.styles_display_block_desktop__1j5zl92a{display:block}.styles_paddingTop_small_desktop__1j5zl92e{padding-top:10px}.styles_paddingTop_medium_desktop__1j5zl92i{padding-top:20px}}
"
`;

exports[`Stylesheet sprinkles - vite should create valid stylesheet 1`] = `".styles_display_flex_mobile__1j5zl920{display:flex}.styles_display_none_mobile__1j5zl924{display:none}.styles_display_block_mobile__1j5zl928{display:block}.styles_paddingTop_small_mobile__1j5zl92c{padding-top:10px}.styles_paddingTop_medium_mobile__1j5zl92g{padding-top:20px}@media screen and (min-width:768px){.styles_display_flex_tablet__1j5zl921{display:flex}.styles_display_none_tablet__1j5zl925{display:none}.styles_display_block_tablet__1j5zl929{display:block}.styles_paddingTop_small_tablet__1j5zl92d{padding-top:10px}.styles_paddingTop_medium_tablet__1j5zl92h{padding-top:20px}}@media screen and (min-width:1024px){.styles_display_flex_desktop__1j5zl922{display:flex}@supports not (display:grid){[data-dark-mode] .styles_display_flex_darkDesktop__1j5zl923{display:flex}[data-dark-mode] .styles_display_none_darkDesktop__1j5zl927{display:none}[data-dark-mode] .styles_display_block_darkDesktop__1j5zl92b{display:block}[data-dark-mode] .styles_paddingTop_small_darkDesktop__1j5zl92f{padding-top:10px}[data-dark-mode] .styles_paddingTop_medium_darkDesktop__1j5zl92j{padding-top:20px}}.styles_display_none_desktop__1j5zl926{display:none}.styles_display_block_desktop__1j5zl92a{display:block}.styles_paddingTop_small_desktop__1j5zl92e{padding-top:10px}.styles_paddingTop_medium_desktop__1j5zl92i{padding-top:20px}}"`;

exports[`Stylesheet sprinkles - webpack should create valid stylesheet 1`] = `
".styles_display_flex_mobile__3nw5tz0 {
  display: flex;
}
.styles_display_none_mobile__3nw5tz4 {
  display: none;
}
.styles_display_block_mobile__3nw5tz8 {
  display: block;
}
.styles_paddingTop_small_mobile__3nw5tzc {
  padding-top: 10px;
}
.styles_paddingTop_medium_mobile__3nw5tzg {
  padding-top: 20px;
}
@media screen and (min-width: 768px) {
  .styles_display_flex_tablet__3nw5tz1 {
    display: flex;
  }
  .styles_display_none_tablet__3nw5tz5 {
    display: none;
  }
  .styles_display_block_tablet__3nw5tz9 {
    display: block;
  }
  .styles_paddingTop_small_tablet__3nw5tzd {
    padding-top: 10px;
  }
  .styles_paddingTop_medium_tablet__3nw5tzh {
    padding-top: 20px;
  }
}
@media screen and (min-width: 1024px) {
  .styles_display_flex_desktop__3nw5tz2 {
    display: flex;
  }
  @supports not (display: grid) {
    [data-dark-mode] .styles_display_flex_darkDesktop__3nw5tz3 {
      display: flex;
    }
    [data-dark-mode] .styles_display_none_darkDesktop__3nw5tz7 {
      display: none;
    }
    [data-dark-mode] .styles_display_block_darkDesktop__3nw5tzb {
      display: block;
    }
    [data-dark-mode] .styles_paddingTop_small_darkDesktop__3nw5tzf {
      padding-top: 10px;
    }
    [data-dark-mode] .styles_paddingTop_medium_darkDesktop__3nw5tzj {
      padding-top: 20px;
    }
  }
  .styles_display_none_desktop__3nw5tz6 {
    display: none;
  }
  .styles_display_block_desktop__3nw5tza {
    display: block;
  }
  .styles_paddingTop_small_desktop__3nw5tze {
    padding-top: 10px;
  }
  .styles_paddingTop_medium_desktop__3nw5tzi {
    padding-top: 20px;
  }
}

/*# sourceMappingURL=main.css.map*/"
`;

exports[`Stylesheet themed - esbuild should create valid stylesheet 1`] = `
":root,.themes__cvta170{--colors-background-color__cvta171: blue;--colors-text__cvta172: white;--space-1__cvta173: 4px;--space-2__cvta174: 8px;--space-3__cvta175: 12px}.themes__cvta176{--colors-background-color__cvta171: green;--colors-text__cvta172: white;--space-1__cvta173: 8px;--space-2__cvta174: 12px;--space-3__cvta175: 16px}.themes__cvta177{--colors-background-color__cvta171: pink;--colors-text__cvta172: purple;--space-1__cvta173: 6px;--space-2__cvta174: 12px;--space-3__cvta175: 18px}@media screen and (min-width: 768px){.themes__cvta177{--colors-background-color__cvta171: purple;--colors-text__cvta172: pink}}.shared__4dtfen0{box-shadow:0 0 5px red}body{background-color:#87ceeb}@font-face{src:local(\\"Impact\\");font-family:\\"styles__jteyb10\\"}@font-face{src:local(\\"Comic Sans MS\\");font-family:MyGlobalComicSans}@keyframes styles__jteyb11{0%{transform:translateY(-4px)}to{transform:translateY(4px)}}@keyframes globalSlide{0%{transform:translateY(-4px)}to{transform:translateY(4px)}}.styles__jteyb12{animation:3s infinite alternate globalSlide ease-in-out;display:flex;flex-direction:column;gap:var(--space-2__cvta174);padding:var(--space-3__cvta175)}.styles__jteyb13{animation:3s infinite alternate styles__jteyb11 ease-in-out;font-family:\\"styles__jteyb10\\";background-color:var(--colors-background-color__cvta171, \\"THIS FALLBACK VALUE SHOULD NEVER BE USED\\");color:var(--colors-text__cvta172)}.themes__cvta176 .themes__cvta170 .styles__jteyb12 .styles__jteyb13{font-family:MyGlobalComicSans;outline:5px solid red}html .styles_1\\\\/2__jteyb16{opacity:var(--jteyb14, .5)}html .styles_1\\\\/4__jteyb17{opacity:var(--jteyb14, var(--jteyb15, .25))}@media only screen and (min-width: 500px){.styles__jteyb12{border:1px solid var(--colors-background-color__cvta171)}.styles__jteyb13{border-radius:9999px}}
"
`;

exports[`Stylesheet themed - vite should create valid stylesheet 1`] = `".themes__cvta170,:root{--colors-background-color__cvta171:blue;--colors-text__cvta172:white;--space-1__cvta173:4px;--space-2__cvta174:8px;--space-3__cvta175:12px}.themes__cvta176{--colors-background-color__cvta171:green;--colors-text__cvta172:white;--space-1__cvta173:8px;--space-2__cvta174:12px;--space-3__cvta175:16px}.themes__cvta177{--colors-background-color__cvta171:pink;--colors-text__cvta172:purple;--space-1__cvta173:6px;--space-2__cvta174:12px;--space-3__cvta175:18px}@media screen and (min-width:768px){.themes__cvta177{--colors-background-color__cvta171:purple;--colors-text__cvta172:pink}}.shared__4dtfen0{box-shadow:0 0 5px red}body{background-color:#87ceeb}@font-face{src:local(\\"Impact\\");font-family:styles__jteyb10}@font-face{src:local(\\"Comic Sans MS\\");font-family:MyGlobalComicSans}@keyframes styles__jteyb11{0%{transform:translateY(-4px)}100%{transform:translateY(4px)}}@keyframes globalSlide{0%{transform:translateY(-4px)}100%{transform:translateY(4px)}}.styles__jteyb12{animation:3s infinite alternate globalSlide ease-in-out;display:flex;flex-direction:column;gap:var(--space-2__cvta174);padding:var(--space-3__cvta175)}.styles__jteyb13{animation:3s infinite alternate styles__jteyb11 ease-in-out;font-family:styles__jteyb10;background-color:var(--colors-background-color__cvta171, \\"THIS FALLBACK VALUE SHOULD NEVER BE USED\\");color:var(--colors-text__cvta172)}.themes__cvta176 .themes__cvta170 .styles__jteyb12 .styles__jteyb13{font-family:MyGlobalComicSans;outline:5px solid red}html .styles_1\\\\/2__jteyb16{opacity:var(--jteyb14,.5)}html .styles_1\\\\/4__jteyb17{opacity:var(--jteyb14,var(--jteyb15,.25))}@media only screen and (min-width:500px){.styles__jteyb12{border:1px solid var(--colors-background-color__cvta171)}.styles__jteyb13{border-radius:9999px}}"`;

exports[`Stylesheet themed - webpack should create valid stylesheet 1`] = `
":root, .themes_theme__cdwe0v0 {
  --colors-background-color__cdwe0v1: blue;
  --colors-text__cdwe0v2: white;
  --space-1__cdwe0v3: 4px;
  --space-2__cdwe0v4: 8px;
  --space-3__cdwe0v5: 12px;
}
.themes_altTheme__cdwe0v6 {
  --colors-background-color__cdwe0v1: green;
  --colors-text__cdwe0v2: white;
  --space-1__cdwe0v3: 8px;
  --space-2__cdwe0v4: 12px;
  --space-3__cdwe0v5: 16px;
}
.themes_responsiveTheme__cdwe0v7 {
  --colors-background-color__cdwe0v1: pink;
  --colors-text__cdwe0v2: purple;
  --space-1__cdwe0v3: 6px;
  --space-2__cdwe0v4: 12px;
  --space-3__cdwe0v5: 18px;
}
@media screen and (min-width: 768px) {
  .themes_responsiveTheme__cdwe0v7 {
    --colors-background-color__cdwe0v1: purple;
    --colors-text__cdwe0v2: pink;
  }
}
.shared_shadow__5iqv0q0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: skyblue;
}
@font-face {
  src: local(\\"Impact\\");
  font-family: \\"styles_impact__blj6yb0\\";
}
@font-face {
  src: local(\\"Comic Sans MS\\");
  font-family: MyGlobalComicSans;
}
@keyframes styles_slide__blj6yb1 {
  0% {
    transform: translateY(-4px);
  }
  100% {
    transform: translateY(4px);
  }
}
@keyframes globalSlide {
  0% {
    transform: translateY(-4px);
  }
  100% {
    transform: translateY(4px);
  }
}
.styles_container__blj6yb2 {
  animation: 3s infinite alternate globalSlide ease-in-out;
  display: flex;
  flex-direction: column;
  gap: var(--space-2__cdwe0v4);
  padding: var(--space-3__cdwe0v5);
}
.styles_button__blj6yb3 {
  animation: 3s infinite alternate styles_slide__blj6yb1 ease-in-out;
  font-family: \\"styles_impact__blj6yb0\\";
  background-color: var(--colors-background-color__cdwe0v1, \\"THIS FALLBACK VALUE SHOULD NEVER BE USED\\");
  color: var(--colors-text__cdwe0v2);
}
.themes_altTheme__cdwe0v6 .themes_theme__cdwe0v0 .styles_container__blj6yb2 .styles_button__blj6yb3 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
html .styles_opacity_1\\\\/2__blj6yb6 {
  opacity: var(--undefined-var1__blj6yb4, 0.5);
}
html .styles_opacity_1\\\\/4__blj6yb7 {
  opacity: var(--undefined-var1__blj6yb4, var(--undefined-var2__blj6yb5, 0.25));
}
@media only screen and (min-width: 500px) {
  .styles_container__blj6yb2 {
    border: 1px solid var(--colors-background-color__cdwe0v1);
  }
  .styles_button__blj6yb3 {
    border-radius: 9999px;
  }
}

/*# sourceMappingURL=main.css.map*/"
`;

exports[`Stylesheet unused-modules - esbuild should create valid stylesheet 1`] = `
"body{background-color:#e6e6fa}body{border:5px solid black}.shared__fbihc60{display:flex}.used__36o3zb0{height:100px;width:100px;background:green}.reset__1j78ztq0{box-sizing:border-box}
"
`;

exports[`Stylesheet unused-modules - vite should create valid stylesheet 1`] = `"body{background-color:#e6e6fa}body{border:5px solid #000}.shared__fbihc60{display:flex}.unused__g6ndup0{color:red;background-color:#00f}.used__36o3zb0{height:100px;width:100px;background:green}.reset__1j78ztq0{box-sizing:border-box}"`;

exports[`Stylesheet unused-modules - webpack should create valid stylesheet 1`] = `
"body {
  background-color: lavender;
}
.reset_default__13dn9qt0 {
  box-sizing: border-box;
}
body {
  border: 5px solid black;
}
.shared_default__1mtonhh0 {
  display: flex;
}
.used_className__plqi3i0 {
  height: 100px;
  width: 100px;
  background: green;
}

/*# sourceMappingURL=main.css.map*/"
`;
